tests/mir-opt/issue_41110.test.ElaborateDrops.after.mir

```diff
@@ -16,14 +17,17 @@ fn test() -> () {
     }
 
     bb0: {
+        _7 = const false;                // scope 0 at $DIR/issue_41110.rs:+1:9: +1:10
         _6 = const false;                // scope 0 at $DIR/issue_41110.rs:+1:9: +1:10
         StorageLive(_1);                 // scope 0 at $DIR/issue_41110.rs:+1:9: +1:10
-        _6 = const true;                 // scope 0 at $DIR/issue_41110.rs:+1:13: +1:14
+        _7 = const true;                 // scope 0 at $DIR/issue_41110.rs:+1:13: +1:14
         _1 = S;                          // scope 0 at $DIR/issue_41110.rs:+1:13: +1:14
         StorageLive(_2);                 // scope 1 at $DIR/issue_41110.rs:+2:9: +2:14
+        _6 = const true;                 // scope 1 at $DIR/issue_41110.rs:+2:17: +2:18
         _2 = S;                          // scope 1 at $DIR/issue_41110.rs:+2:17: +2:18
         StorageLive(_3);                 // scope 2 at $DIR/issue_41110.rs:+3:5: +3:12
         StorageLive(_4);                 // scope 2 at $DIR/issue_41110.rs:+3:10: +3:11
+        _6 = const false;                // scope 2 at $DIR/issue_41110.rs:+3:10: +3:11
```

Comment:
Having multiple assignments to the drop flags is a bit surprising. Where does it come from? Is there a way to avoid this?